src/video/wayland/SDL_waylandwindow.c

```diff
@@ -2029,11 +2002,6 @@ void Wayland_SetWindowSize(_THIS, SDL_Window * window)
     wind->floating_width = window->windowed.w;
     wind->floating_height = window->windowed.h;
 
-    region = wl_compositor_create_region(data->compositor);
-    wl_region_add(region, 0, 0, window->w, window->h);
-    wl_surface_set_opaque_region(wind->surface, region);
-    wl_region_destroy(region);
```

Comment:
Same as the other set_opaque_region comment

Reply:
It's not clear from the diff since it's a bit truncated, but in both cases there is a call to ConfigureWindowGeometry(), which is where the opaque region setting moved to, right after where this used to be and before anything is committed.  Unless you specifically mean that this must happen before the xdg_surface_set_window_geometry() call due to some compositor issue?  Both wl_surface_set_opaque_region() and xdg_surface_set_window_geometry() are buffered and committed atomically when the surface commit call is made, so the order shouldn't matter, but if it does, I can change this back.

Reply:
Oh, in that case I think it should be okay - going to CC @dos1 just in case, though.